replica/replica.go

```diff
@@ -825,11 +851,20 @@ func (r *Replica) createDisk(name string, userCreated bool, created string) erro
 
 		r.updateChildDisk(oldHead, newSnapName)
 		r.activeDiskData[len(r.activeDiskData)-1].Name = newSnapName
+		r.diskList = r.diskList[:len(r.diskList)-1]
+		r.diskList = append(r.diskList, newSnapName)
 	}
 	delete(r.diskData, oldHead)
+	//delete(r.diskList, oldHead)
 
 	r.info = info
 	r.volume.files = append(r.volume.files, f)
+	r.volume.ReadOnlyIndx = append(r.volume.ReadOnlyIndx, false)
+	r.diskList = append(r.diskList, newHeadDisk.Name)
+	if userCreated {
```

Comment:
why `- 2`? userCreated snapshot will be just next to the head right? so, should be `- 1`?

Reply:
do we need to lock protect this?

Reply:
Lock protection is not required as all the functions which are using it are lock protected.